### ISSUE TITLE
glibc's has easily removable dependency for bash in its scripts.

### ISSUE THREAD

**Issue by Arthur2e5** [AOSC-Dev/aosc-os-core#5]
glibc's scripts all uses `#! /bin/bash` as the she-bang, to make use of the `$"msgid"` feature in bash.

Though such strings are non-POSIX (as in my memory) and may cause weird results, all the shells we provide plus busybox `ash` `hush` work well with the `$SHELL $(which ldd) /bin/bash` test. (zsh needs to `emulate -c sh` first, but it does it automatically when finding its `*argv` to be `(.*/)*sh`).

Therefore, it would be mostly safe to change it to `#!/bin/sh` so it can still go well without the `bash` dependency literally. However, I need someone to test if `LC_ALL=zh_CN.UTF-8 sh /bin/ldd` loses those gettext translations. If they are lost, we are not changing the shebang or we are making bugs.

Anyway, this may not be quite necessary for OS3 since we always have bash ready.

By the way, `ldd` in glibc2 supported `-V`, which does the same as `--version`. This can be easily implemented with in that `case` and maybe we can patch it in the same time, although there is no egg use.

**Comment by MingcongBai** [comment]
> However, I need someone to test if LC_ALL=zh_CN.UTF-8 sh /bin/ldd loses those gettext translations. If they are lost, we are not changing the shebang or we are making bugs.

No, changing to `#!/bin/sh` would not take gettext translation away.

> Anyway, this may not be quite necessary for OS3 since we always have bash ready.

For the sake of logic, let's do this.

> By the way, ldd in glibc2 supported -V, which does the same as --version. This can be easily implemented with in that case and maybe we can patch it in the same time, although there is no egg use.

There is no need for intervention, I believe.

**Comment by MingcongBai** [comment]
```
/usr/bin/sotruss: Bourne-Again shell script, ASCII text executable
/usr/bin/ldd: POSIX shell script, ASCII text executable
/usr/bin/tzselect: Bourne-Again shell script, ASCII text executable
/usr/bin/catchsegv: POSIX shell script, ASCII text executable
/usr/bin/xtrace: Bourne-Again shell script, ASCII text executable
```

@Arthur2e5 Let's get the testing going shall we?

First of all, what to test, except for `gettext` issue.

**Comment by Arthur2e5** [comment]
@MingcongBai 
1. File can be wrong even for those cases, so trust me and use `head -n 1`.
2. Gettext is the only issue.

**Comment by MingcongBai** [comment]
@Arthur2e5 can you further explain point 1?

**Comment by MingcongBai** [comment]
@Arthur2e5 can you further explain point 1?

**Comment by Arthur2e5** [comment]
```
180bot root [ bin ] # head -n 1 sotruss ldd tzselect catchsegv xtrace
==> sotruss <==
#! /bin/bash

==> ldd <==
#! /bin/bash

==> tzselect <==
#!/bin/bash

==> catchsegv <==
#! /bin/sh

==> xtrace <==
#! /bin/bash
```

**Comment by MingcongBai** [comment]
I will be holding this change, is this okay with you?
